tests/test_transforms/test_integration.py

```diff
@@ -0,0 +1,239 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+import torch
+import torch.nn as nn
+from compressed_tensors.transforms import Hadamard, RandomHadamard, Transforms
+from compressed_tensors.transforms.transform_args import (
+    ModuleTarget,
+    TransformationArgs,
+)
+from compressed_tensors.transforms.transform_config import TransformationConfig
+from compressed_tensors.transforms.transform_data import TransformData
+from compressed_tensors.transforms.transform_scheme import TransformationScheme
+
+
+@pytest.fixture
+def transform_recipe_basic():
+    targets = ["Linear"]
+    module_targets = [ModuleTarget.WEIGHTS]
+    linear_layer_args = TransformationArgs(
+        targets=targets, module_targets=module_targets
+    )
+
+    scheme = TransformationScheme(
+        transform_type="hadamard",
+        groups=[linear_layer_args],
+        transform_creation_args={"size": 64},
```

Comment:
Rather than having a field for `transform_type` and another for the kwargs, can't we just do this?
```python
transform={"type":"hadamard", "size" :64}
```
```python
transform={"type":"scalar", "scalar_val": 2.5}
```
This is a sort of polymorphic deserialization (the resultant object will be of type `Hadamard` or `ScalarMultiply`) is doable in Pydantic v2 as far as i can tell. If we want to try to adopt it I can try to create a PR for it

Reply:
So you would pass in the dictionary when creating a scheme?
e.g. TransformsSchemes(transform={"type":"scalar", "scalar_val": 2.5})?
And then have the model do internal validation?

I guess what would be the benefit over having a separate key field?

Reply:
Just a cleaner data model, args are better scoped, and it removes the need for custom deserialization logic, it's just handled by Pydantic. Nothing like this would be needed:
```python
TransformRegistry.load(scheme["transform_type"]).create(**scheme["transform_creation_args"])
```